CIME/SystemTests/system_tests_common.py

```diff
@@ -225,14 +225,15 @@ def _set_restart_interval(
             smon = startdatetime.month
             ryr = restdatetime.year
             rmon = restdatetime.month
-            while ryr > syr:
-                if rmon > 2 and calendar.isleap(ryr):
+            while ryr > syr and (ryr < restdatetime.year or rmon > 2):
```

Comment:
Since this has to loop over every year from startdate to restdate I also wonder about how this would do for an especially long period of many years. Practically most likely we aren't going to do restart tests for especially long periods, but for T compsets for example multi-decade restart tests could make sense.

Reply:
I agree that unit tests would be ideal here - covering a few specific cases - if they aren't too hard to put in place. If they are hard to put in place for some reason, then having some comments explaining this logic feels like an acceptable alternative (actually, whether or not there are unit tests).